presto-spark-base/src/main/java/com/facebook/presto/spark/PrestoSparkMemoryBasedBroadcastDependency.java

```diff
@@ -81,6 +85,32 @@ public Broadcast<List<PrestoSparkSerializedPage>> executeBroadcast(JavaSparkCont
         return broadcastVariable;
     }
 
+    private long getNativeInMemoryBytes(List<PrestoSparkSerializedPage> prestoSparkSerializedPages, PagesSerde pagesSerde)
+    {
+        return prestoSparkSerializedPages.stream()
+                .map(prestoSparkSerializedPage -> PrestoSparkUtils.toSerializedPage(prestoSparkSerializedPage))
+                .map(serializedPage -> pagesSerde.deserialize(serializedPage))
+                // Extract fileSize.
```

Comment:
Is this file size or sum of "retainedSize" of the vectors? Does this code run on the producer or consumer?

Reply:
Its sum of "retainedSize". will update the comment to reflect "estimateFlatSize".
This code runs on driver. We can say its consumer but actual consumers are executors/native process running on executors.